internal/v2/controller/http/callback.go

```diff
@@ -0,0 +1,120 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//
+// Copyright (C) 2020 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package http
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/edgexfoundry/device-sdk-go/internal/common"
+	"github.com/edgexfoundry/device-sdk-go/internal/v2/application"
+	"github.com/edgexfoundry/go-mod-core-contracts/errors"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/dtos/requests"
+	"github.com/gorilla/mux"
+)
+
+func (c *V2HttpController) DeleteDevice(writer http.ResponseWriter, request *http.Request) {
+	vars := mux.Vars(request)
+	id := vars[common.IdVar]
+	correlationID := request.Header.Get(common.CorrelationHeader)
+
+	err := application.DeleteDevice(id, c.dic)
+	if err == nil {
+		c.sendResponse(writer, request, common.APIV2DeviceCallbackIdRoute, nil, http.StatusOK)
+	} else {
+		c.sendEdgexError(writer, request, err, common.APIV2DeviceCallbackIdRoute, correlationID)
+	}
+}
+
+func (c *V2HttpController) AddUpdateDevice(writer http.ResponseWriter, request *http.Request) {
+	switch request.Method {
+	case http.MethodPost:
+		c.addDevice(writer, request)
+	case http.MethodPut:
+		c.updateDevice(writer, request)
+	}
+}
```

Comment:
This can be removed if routes are update to use `addDevice` & `updateDevice` once they are exposed.

```
	c.addReservedRoute(sdkCommon.APIV2DeviceCallbackRoute, c.v2HttpController.AddDevice).Methods(http.MethodPost)
	c.addReservedRoute(sdkCommon.APIV2DeviceCallbackRoute, c.v2HttpController.UpdateDevice).Methods(http.MethodPut)
```